src/message.rs

```diff
@@ -61,15 +61,38 @@ impl From<crate::asynk::Message> for Message {
 }
 
 impl Message {
+    /// Creates new empty `Message`, without a Client.
+    /// Useful for passing `Message` data or creating `Message` instance without caring about `Client`,
+    /// but cannot be used on it's own for associated methods as those require `Client` injected into `Message`
+    /// and will error without it.
+    pub fn new(
+        subject: &str,
+        reply: Option<&str>,
+        data: impl AsRef<[u8]>,
+        headers: Option<Headers>,
+    ) -> Message {
+        Message {
+            subject: subject.to_string(),
+            reply: reply.map(String::from),
+            data: data.as_ref().to_vec(),
+            headers,
+            ..Default::default()
+        }
+    }
+
     /// Respond to a request message.
     pub fn respond(&self, msg: impl AsRef<[u8]>) -> io::Result<()> {
-        match self.reply.as_ref() {
-            None => Err(io::Error::new(
-                io::ErrorKind::InvalidInput,
-                "no reply subject available",
-            )),
-            Some(reply) => self.client.publish(reply, None, None, msg.as_ref()),
-        }
+        let reply = self.reply.as_ref().ok_or_else(|| {
+            io::Error::new(io::ErrorKind::InvalidInput, "No reply subject to reply to")
+        })?;
+        let client = self.client.as_ref().ok_or_else(|| {
+            io::Error::new(
+                io::ErrorKind::NotConnected,
+                "no connected client to reply with",
```

Comment:
Let's change to "message not bound to a client". For this one and the one below.

Reply:
done, also extracted `&str` to const as it occured 3 times in code already.